### PR TITLE
initiate many-to-many relationship for users and stocks

### PR DESCRIPTION
**Pull request by LeeRenJie**
depends on #124

### CODE REVIEW

File: finance-tracker/config/routes.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
+  resources :user_stocks
```
Comment (tlkiong) [inline finance-tracker/config/routes.rb:2]: why at the top of router file?

**Review by tlkiong — CHANGES_REQUESTED** [review]

**Review by tlkiong — CHANGES_REQUESTED** [review]
Why do you need to have `UserStock` model?
*if you want to tell me its required, you can try removing the model file (keep the migration). Then change the has_many user_stocks & has_many xx through user_stocks with:

```
has_and_belongs_to_many xx
```

**Comment by LeeRenJie** [comment]
> Why do you need to have UserStock model?

initiate many-to-many relationships between users and stocks through this table. A user can track many stocks, a stock can be tracked by many different users

**Comment by LeeRenJie** [comment]
> Why do you need to have `UserStock` model?
> *if you want to tell me its required, you can try removing the model file (keep the migration). Then change the has_many user_stocks & has_many xx through user_stocks with:
> 
> ```
> has_and_belongs_to_many xx
> ```

We need UserStock model to initiate the many-to-many association through this table with `has_many xx through`. Because it is better compared to `has_and_belongs_to_many` since we can add more columns to the join table in the future.

**Review by tlkiong — APPROVED** [review]
Very good.
Never ever initiate a many to many relationship without the join model.

**Pull request merged**